cmd/cwa-allocator/target/discovery.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package target
+
+import (
+	"hash"
+	"hash/fnv"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/model/relabel"
+	"gopkg.in/yaml.v3"
+
+	allocatorWatcher "github.com/aws/amazon-cloudwatch-agent-operator/cmd/cwa-allocator/watcher"
+)
+
+var (
+	targetsDiscovered = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "opentelemetry_allocator_targets",
```

Comment:
Should this be `amazoncloudwatchagent_allocator_targets`?

Reply:
Oops yes, my regex failed :/